docs/decisions/0006-middleware-support-for-http-clients.rst

```diff
@@ -0,0 +1,40 @@
+middleware for HTTP client
+==========================================
+
+Status
+------
+
+
+
+Context
+-------
+
+We currently expose a HTTP client(axios instance) via ``getAuthenticatedHttpClient`` and ``getHttpClient`` used in
+our MFEs to make HTTP requests. Since edX uses Django for its backend services, the MFEs have to handle snake-cased <-> camelCase
+conversions manually. The 'axios-case-converter' package provides a middleware that would handle the conversion automatically. In the
+process of integrating the package, we thought it might be beneficial to
+establish a pattern for supporting middleware.
+
+Decision
+--------
+
+``getAuthenticatedHttpClient`` and ``getHttpClient`` take an ``options`` argument that includes configuration for the client.
+Only ``useCache`` is currently supported. We will introduce a new option ``middleware``, which will be a list of middleware
+that will be applied to the base axios instance. Each item in the list can either be a function or an array in the form of [function, options].
+We will apply the middleware and handle any exceptions that occur.
+
+It will look like this:
+    {
+        useCache: true,
+        middleware: [
+            axiosConvertCase,
+            [axiosRetry, { retries: 3 }],
+            ...
+        ]
+    }
+
+We will export any middleware functions that we support, i.e. ``applyCaseMiddleware`` from frontend-platform.
```

Comment:
I believe one of the benefits of this middleware approach is that consumers could write their own middleware functions. `@edx/frontend-platform` might export official middleware as pass-thrus (e.g., `applyCaseMiddleware`)  but theoretically consumers could use others or create their own outside of ones exported by `@edx/frontend-platform`.

In this sense, is the "... that we support..." clause accurate?

Reply:
yeah I'll rephrase this part